### PR TITLE
Hooks

### PR DESCRIPTION
**Pull request by Pzixel**
First attempt to add hooks. Don't know it it correctly determines path to `rustfmt` on Linux. Can you check it @nebgnahz, please? 

I'm also waiting for commiting our C rules to make it happen too.

### CODE REVIEW

File: setup_hooks.sh [hunk 1]
```diff
@@ -0,0 +1,10 @@
+#!/bin/sh
+rustup component add rustfmt-preview
+
+rustfmt_path=`find $HOME/.rustup/toolchains/ -wholename '*bin/rustfmt*'`
```
Comment (nebgnahz) [inline setup_hooks.sh]: `which rustfmt` should suffice?

File: setup_hooks.sh [hunk 2]
```diff
@@ -0,0 +1,10 @@
...
+echo "#!/bin/sh
+find . -name *.rs -type f -print | xargs $rustfmt_path"  > .git/hooks/pre-commit
```
Comment (nebgnahz) [inline setup_hooks.sh]: This only finds *.rs in the top leve.

Also `xargs` will just write all the diffs, and the return code is 0, i.e. the commit will pass.
```
$ git add src/lib.rs
$ git commit -m "small tweaks" // succeed despite format issue
$ git status
Changes not staged for commit:
   modified: src/lib.rs
```

We would want the commit to fail if the format is wrong.

File: setup_hooks.sh [hunk 3]
```diff
@@ -0,0 +1,10 @@
+#!/bin/sh
```
Comment (nebgnahz) [inline setup_hooks.sh]: P.S. for this file as an executable, you should also `chmod +x`.

File: setup_hooks.sh (hunk 1 above)
Comment (Pzixel) [reply setup_hooks.sh]: On my machine I have following:
```
SOLO@DESKTOP-0LLRSJV MINGW64 ~/Documents/Github/cv-rs (feature/FLANN)
$ find $HOME/.rustup/toolchains/ -wholename '*rustfmt*'
/c/Users/SOLO/.rustup/toolchains/stable-x86_64-pc-windows-gnu/bin/rustfmt.exe
/c/Users/SOLO/.rustup/toolchains/stable-x86_64-pc-windows-gnu/lib/rustlib/manife                         st-rustfmt-preview-x86_64-pc-windows-gnu
/c/Users/SOLO/.rustup/toolchains/stable-x86_64-pc-windows-gnu/share/doc/rustfmt
/c/Users/SOLO/.rustup/toolchains/stable-x86_64-pc-windows-gnu/share/doc/rustfmt/                         LICENSE-APACHE
/c/Users/SOLO/.rustup/toolchains/stable-x86_64-pc-windows-gnu/share/doc/rustfmt/                         LICENSE-MIT
/c/Users/SOLO/.rustup/toolchains/stable-x86_64-pc-windows-gnu/share/doc/rustfmt/                         README.md
```
So we only want an executable here, which is located in `bin`

File: setup_hooks.sh (hunk 2 above)
Comment (Pzixel) [reply setup_hooks.sh]: This is pre-commit hook, it have nothing with passing build.

The idea is that guy has broken formatting, it writes `git commit` and hook just fixes all issues for him. It's better to fix an issue automatically than blame user and ask him to fix all manually.

**Comment by Pzixel** [comment]
This PR is completely broken, so let's move in #55

**Pull request closed without merging**